tests/unit/s2n_map_test.c

```diff
@@ -19,159 +19,277 @@
 
 #include "api/s2n.h"
 #include "s2n_test.h"
+#include "utils/s2n_map_internal.h"
+
+DEFINE_POINTER_CLEANUP_FUNC(struct s2n_map_iterator *, s2n_map_iterator_free);
 
 int main(int argc, char **argv)
```

Comment:
I don't know why github is rendering this diff in such a confusing way. I modified this file in 2 ways.
1. Added asserts for `s2n_map_size`  in the exisiting `s2n_map_test_case`. I also indented the test case to keep it in it's own scope, which seems to have confused git.
2. Added a new case for `s2n_map_iterator`